server/src/infra/repositories/person.repository.ts

```diff
@@ -96,14 +96,20 @@ export class PersonRepository implements IPersonRepository {
     return this.personRepository.findOne({ where: { id: personId } });
   }
 
-  getByName(userId: string, personName: string): Promise<PersonEntity[]> {
-    return this.personRepository
+  getByName(userId: string, personName: string, withHidden: boolean): Promise<PersonEntity[]> {
+    const queryBuilder = this.personRepository
       .createQueryBuilder('person')
       .leftJoin('person.faces', 'face')
       .where('person.ownerId = :userId', { userId })
-      .andWhere('LOWER(person.name) LIKE :name', { name: `${personName.toLowerCase()}%` })
-      .limit(20)
-      .getMany();
+      .andWhere('LOWER(person.name) LIKE :name', { name: `%${personName.toLowerCase()}%` })
+      .groupBy('person.id')
+      .orderBy('COUNT(face.assetId)', 'DESC')
+      .limit(20);
+
+    if (!withHidden) {
+      queryBuilder.andWhere('person.isHidden = false');
```

Comment:
Do you not need to do `queryBuilder = queryBuilder.andWhere...`?

Reply:
No you don't have to, we use the same pattern in `getAllForUser`
https://github.com/immich-app/immich/blob/250f7fc55c5d118369f06343b0d27c0521cbe7e8/server/src/infra/repositories/person.repository.ts#L76-L82